### PR TITLE
vs2013_fixes

### PR DESCRIPTION
**Pull request by juj**
Build fixes/workarounds to support Visual Studio 2013 build, which has trouble with some new C++11 constructs.

### CODE REVIEW

File: src/passes/CoalesceLocals.cpp [hunk 1]
```diff
@@ -569,7 +569,7 @@ void CoalesceLocalsWithLearning::pickIndices(std::vector<Index>& indices) {
 #ifdef CFG_LEARN_DEBUG
   std::cout << "[learning for " << getFunction()->name << "]\n";
 #endif
-  auto numVars = getFunction()->getNumVars();
+  auto numVars = this->getFunction()->getNumVars();
```
Comment (kripken) [inline src/passes/CoalesceLocals.cpp]: hmm, what is special about this location (and the few others)? what does VS need `this->` here but not in 99% of the other places `this` is implicitly used?

**Comment by dschuff** [comment]
I thought we had decided that we only wanted to support VS2015?

**Comment by juj** [comment]
I have not read about such decision, but I'd like to support VS2013.

**Comment by dschuff** [comment]
I don't know if there was a definitive decision, I'll see if I can dig up the discussion. The short version though is that there's a fairly long list of C++11 features that don't work or are buggy in VS2013, and VS2015 has basically complete working C++11 support, so we can just say "any C++11 feature can be used" instead of worrying about which ones work on which platforms.

**Comment by juj** [comment]
The changes introduced here are small, so could we take a pragmatic approach and support vs2013 as long as that doesn't cause large refactoring? There are users who have asked vs2013 support for binaryen, and given that the current changes are on the complexity of adding few `this->` statements to the build, I would not like to reject the support altogether?

**Comment by kripken** [comment]
Sounds good to me, if the changes are minor like these. If we hit a more serious limitation or a need for large-scale changes though, then I think at that point we should give up on vs2013.

File: src/passes/CoalesceLocals.cpp (hunk 1 above)
Comment (juj) [inline src/passes/CoalesceLocals.cpp]: I'm afraid I have no idea. :/

**Pull request merged**